src/Microsoft.Framework.ApplicationHost2/Program.cs

```diff
@@ -42,22 +51,25 @@ public Task<int> Main(string[] args)
                     return Task.FromResult(exitCode);
                 }
 
-                // Construct the necessary context for hosting the application
-                var builder = new RuntimeHostBuilder(options.ApplicationBaseDirectory);
-
-                // Boot the runtime
-                var host = builder.Build();
-
                 // Check for a project
-                if (host.Project == null)
+                if (!ProjectReader.HasProjectFile(options.ApplicationBaseDirectory))
```

Comment:
I would avoid this IO completely and fail the first time you go to lookup a project. Anything preemptive is bad.